pkg/services/sqlstore/searchstore/search_test.go

```diff
@@ -0,0 +1,215 @@
+// package search_test contains integration tests for search
+package searchstore_test
+
+import (
+	"context"
+	"fmt"
+	"github.com/grafana/grafana/pkg/components/simplejson"
+	"github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/services/sqlstore"
+	"github.com/grafana/grafana/pkg/services/sqlstore/migrator"
+	"github.com/grafana/grafana/pkg/services/sqlstore/permissions"
+	"github.com/grafana/grafana/pkg/services/sqlstore/searchstore"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"testing"
+	"time"
+)
+
+var dialect migrator.Dialect
+
+const (
+	limit int64 = 15
+	page  int64 = 1
+)
+
+func TestBuilder_EqualResults_Basic(t *testing.T) {
+	user := &models.SignedInUser{
+		UserId:  1,
+		OrgId:   1,
+		OrgRole: models.ROLE_EDITOR,
+	}
+
+	db := setupTestEnvironment(t)
+	err := createDashboards(0, 1, user.OrgId)
+	require.NoError(t, err)
+
+	// create one dashboard in another organization that shouldn't
+	// be listed in the results.
+	err = createDashboards(1, 2, 2)
+	require.NoError(t, err)
+
+	builder := &searchstore.Builder{
+		Filters: []interface{}{
+			searchstore.OrgFilter{OrgId: user.OrgId},
+			searchstore.TitleSorter{},
+		},
+		Dialect: dialect,
+	}
+
+	prevBuilder := sqlstore.NewSearchBuilder(user, limit, page, models.PERMISSION_EDIT)
+	prevBuilder.WithDialect(dialect)
+
+	newRes := []sqlstore.DashboardSearchProjection{}
+	err = db.WithDbSession(context.Background(), func(sess *sqlstore.DBSession) error {
+		sql, params := builder.ToSql(limit, page)
+		return sess.SQL(sql, params...).Find(&newRes)
+	})
+	require.NoError(t, err)
+
+	oldRes := []sqlstore.DashboardSearchProjection{}
+	err = db.WithDbSession(context.Background(), func(sess *sqlstore.DBSession) error {
+		sql, params := prevBuilder.ToSql()
+		return sess.SQL(sql, params...).Find(&oldRes)
+	})
+	require.NoError(t, err)
+
+	assert.Len(t, newRes, 1)
```

Comment:
Expectation comes first:
```suggestion
	assert.Len(t, 1, newRes)
```